x/merkledb/proof.go

```diff
@@ -183,19 +181,25 @@ func (proof *Proof) Verify(
 	return nil
 }
 
+type RangeProofMarshaler struct{}
```

Comment:
nit: usually struct definitions are adjacent to receiver definitions. Can we move this below the below `type RangeProof ChangeProof` definition?